test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -5032,59 +5062,26 @@ def revert_to_defaults(self, reverthooks=True, revertqueues=True,
                 reverthooks = True
             else:
                 reverthooks = False
-            hooks = self.status(HOOK, level=logging.DEBUG)
-            hooks = [h['id'] for h in hooks]
-            for h in skip_site_hooks:
-                if h in hooks:
-                    hooks.remove(h)
-            if len(hooks) > 0:
-                self.manager(MGR_CMD_DELETE, HOOK, id=hooks)
+            self.delete_hooks()
         if delqueues:
             revertqueues = False
-            queues = self.status(QUEUE, level=logging.DEBUG)
-            queues = [q['id'] for q in queues]
-            if len(queues) > 0:
-                try:
-                    nodes = self.status(VNODE, logerr=False)
-                    for node in nodes:
-                        if 'queue' in node.keys():
-                            self.manager(MGR_CMD_UNSET, NODE, 'queue',
-                                         node['id'])
-                except:
-                    pass
-                self.manager(MGR_CMD_DELETE, QUEUE, id=queues)
+            self.delete_queues()
             a = {ATTR_qtype: 'Execution',
                  ATTR_enable: 'True',
                  ATTR_start: 'True'}
             self.manager(MGR_CMD_CREATE, QUEUE, a, id='workq')
             setdict.update({ATTR_dfltque: 'workq'})
         if delscheds:
-            self.manager(MGR_CMD_LIST, SCHED)
-            for name in self.schedulers.keys():
-                if name != 'default':
-                    self.schedulers[name].terminate()
-                    sched_log = self.schedulers[
-                        name].attributes['sched_log']
-                    sched_priv = self.schedulers[
-                        name].attributes['sched_priv']
-                    self.du.rm(path=sched_log, sudo=True,
-                               recursive=True, force=True)
-                    self.du.rm(path=sched_priv, sudo=True,
-                               recursive=True, force=True)
-                    self.manager(MGR_CMD_DELETE, SCHED, id=name)
+            self.delete_sched()
 
         if delnodes:
-            try:
-                self.manager(MGR_CMD_DELETE, VNODE, id="@default")
-            except PbsManagerError as e:
-                if "Unknown node" not in e.msg[0]:
-                    raise
+            self.delete_nodes()
         if reverthooks:
             if self.platform == 'cray' or self.platform == 'craysim':
```

Comment:
shouldn't this be part of delete_hooks?

Reply:
For me it looks like setter part in revert to defaults here. I have separated delete configurations from this method

Reply:
but test calls delete_hooks then it will delete these hooks which it shouldn't right... if we don't put this in delete_hooks then test writer has to take care of restoring this hook...

Reply:
Thanks I got it now.